compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java

```diff
@@ -3436,11 +3437,38 @@ public BLangNode transform(ArrayTypeDescriptorNode arrayTypeDescriptorNode) {
             } else {
                 Node keyExpr = arrayTypeDescriptorNode.arrayLength().get();
                 if (keyExpr.kind() == SyntaxKind.NUMERIC_LITERAL) {
+                    int length = 0;
                     Token literalToken = ((BasicLiteralNode) keyExpr).literalToken();
                     if (literalToken.kind() == SyntaxKind.DECIMAL_INTEGER_LITERAL_TOKEN) {
-                        sizes.add(new BLangLiteral(Integer.parseInt(literalToken.text()), symTable.intType));
+                        double lengthCheck = Double.parseDouble(literalToken.text());
```

Comment:
Can't we use long instead and catch exceptions instead? Then we probably won't need the custom logic from L3455 also.

Reply:
okay. I will use that.